File: src/main/java/hzt/aoc/day21/Food.java
package hzt.aoc.day21;

import hzt.collections.MutableSetX;
import hzt.collections.SetX;

import java.util.Collections;
import java.util.Set;

public class Food {

    private final Set<String> ingredients;
    private final Set<String> allergens;

    public Food(Set<String> ingredients, Set<String> allergens) {
        this.ingredients = ingredients;
        this.allergens = allergens;
    }

    public MutableSetX<String> getIngredients() {
        return MutableSetX.of(ingredients);
    }

    public Set<String> getAllergens() {
        return Collections.unmodifiableSet(allergens);
    }

    @Override
    public String toString() {
        return "Food{" +
                "ingredients=" + ingredients +
                ", allergens=" + allergens +
                '}';
    }
}

File: src/main/java/hzt/aoc/day21/Food.kt
package hzt.aoc.day21

import java.util.*

class Food(private val ingredients: Set<String>, private val allergens: Set<String>) {
    fun getIngredients(): Set<String> {
        return Collections.unmodifiableSet(ingredients)
    }

    fun getAllergens(): Set<String> {
        return Collections.unmodifiableSet(allergens)
    }

    override fun toString(): String {
        return "Food{" +
                "ingredients=" + ingredients +
                ", allergens=" + allergens +
                '}'
    }
}
